### PR TITLE
[SPARK-8567] [SQL] Increase the timeout of HiveSparkSubmitSuite

### PR DESCRIPTION
**Pull request by yhuai**
https://issues.apache.org/jira/browse/SPARK-8567

### CODE REVIEW

**Comment by JoshRosen** [comment]
Jenkins, retest this please.

**Comment by yhuai** [comment]
hmm... Jenkins still does not have the logs of applications launched in `HiveSparkSubmitSuite`.

**Comment by JoshRosen** [comment]
@yhuai, where are those logs written when you run the tests locally?  We might have to add those log paths to the list that Jenkins checks when attaching build artifacts in the post-build step.

**Comment by yhuai** [comment]
They are in `sql/hive/target/uni-tests.log`. The log in jenkins only shows
```
12:47:58.306 pool-1-thread-1-ScalaTest-running-HiveSparkSubmitSuite INFO HiveSparkSubmitSuite: 

===== TEST OUTPUT FOR o.a.s.sql.hive.HiveSparkSubmitSuite: 'SPARK-8368: includes jars passed in through --jars' =====

12:49:04.038 pool-1-thread-1-ScalaTest-running-HiveSparkSubmitSuite INFO HiveSparkSubmitSuite: 

===== FINISHED o.a.s.sql.hive.HiveSparkSubmitSuite: 'SPARK-8368: includes jars passed in through --jars' =====

12:49:04.039 pool-1-thread-1-ScalaTest-running-HiveSparkSubmitSuite INFO HiveSparkSubmitSuite: 

===== TEST OUTPUT FOR o.a.s.sql.hive.HiveSparkSubmitSuite: 'SPARK-8020: set sql conf in spark conf' =====

12:49:52.939 pool-1-thread-1-ScalaTest-running-HiveSparkSubmitSuite INFO HiveSparkSubmitSuite: 

===== FINISHED o.a.s.sql.hive.HiveSparkSubmitSuite: 'SPARK-8020: set sql conf in spark conf' =====

12:49:52.941 pool-1-thread-1-ScalaTest-running-HiveSparkSubmitSuite INFO HiveSparkSubmitSuite: 

===== TEST OUTPUT FOR o.a.s.sql.hive.HiveSparkSubmitSuite: 'SPARK-8489: MissingRequirementError during reflection' =====

12:50:42.262 pool-1-thread-1-ScalaTest-running-HiveSparkSubmitSuite INFO HiveSparkSubmitSuite: 

===== FINISHED o.a.s.sql.hive.HiveSparkSubmitSuite: 'SPARK-8489: MissingRequirementError during reflection' =====
```

**Comment by andrewor14** [comment]
I just ran this locally and my `sql/hive/target/unit-test.logs` looks different. Looks like they are logged to the right file there.
```
===== TEST OUTPUT FOR o.a.s.sql.hive.HiveSparkSubmitSuite: 'SPARK-8368: includes jars passed in through --jars' ===== 

14:21:43.037 pool-1-thread-1-ScalaTest-running-HiveSparkSubmitSuite INFO SparkContext: Running Spark version 1.5.0-SNAPSHOT
14:21:43.074 pool-1-thread-1-ScalaTest-running-HiveSparkSubmitSuite DEBUG MutableMetricsFactory: field org.apache.hadoop.metrics2.lib.MutableRate org.apache.hadoop.security.UserGroupInformation$UgiMetrics.loginSuccess with annotation @org.apache.hadoop.metrics2.annotation.Metric(valueName=Time, about=, value=[Rate of successful kerberos logins and latency (milliseconds)], always=false, type=DEFAULT, sampleName=Ops)
14:21:43.078 pool-1-thread-1-ScalaTest-running-HiveSparkSubmitSuite DEBUG MutableMetricsFactory: field org.apache.hadoop.metrics2.lib.MutableRate org.apache.hadoop.security.UserGroupInformation$UgiMetrics.loginFailure with annotation @org.apache.hadoop.metrics2.annotation.Metric(valueName=Time, about=, value=[Rate of failed kerberos logins and latency (milliseconds)], always=false, type=DEFAULT, sampleName=Ops)
14:21:43.078 pool-1-thread-1-ScalaTest-running-HiveSparkSubmitSuite DEBUG MetricsSystemImpl: UgiMetrics, User and group related metrics
14:21:43.305 pool-1-thread-1-ScalaTest-running-HiveSparkSubmitSuite DEBUG KerberosName: Kerberos krb5 configuration not found, setting default realm to empty 
14:21:43.307 pool-1-thread-1-ScalaTest-running-HiveSparkSubmitSuite DEBUG Groups:  Creating new Groups object
...
```

**Comment by yhuai** [comment]
@andrewor14 Yeah, I also got that in my laptop.

**Comment by liancheng** [comment]
Partly reproduced this issue on my laptop. Trying my luck with #6978 by printing rather than logging stdout/stderr of the spawned process.

**Comment by andrewor14** [comment]
Ok, let's retest this please now that #6978 is merged.

**Comment by andrewor14** [comment]
retest this please

File: sql/hive/src/main/scala/org/apache/spark/sql/hive/HiveContext.scala [hunk 1]
```diff
@@ -563,7 +563,7 @@ private[hive] object HiveContext {
     doc = "TODO")
 
   /** Constructs a configuration for hive, where the metastore is located in a temp directory. */
-  def newTemporaryConfiguration(): Map[String, String] = {
+  def newTemporaryConfiguration(useInMemoryDerby: Boolean): Map[String, String] = {
```
Comment (andrewor14) [inline sql/hive/src/main/scala/org/apache/spark/sql/hive/HiveContext.scala]: could just make this default to false

**Comment by andrewor14** [comment]
LGTM as a temporary fix. Let's merge this into master and 1.4 and see if it fixes the test failures.

**Comment by andrewor14** [comment]
I will leave the JIRA open for a day or two until we have verified that this fixes it.

**Comment by liancheng** [comment]
@yhuai As for why the log doesn't appear on Jenkins, my guess is that the `spark-submit` process wasn't started properly because of, for example, some class not found error. I'd suggest to enable `SPARK_PRINT_LAUNCH_COMMAND`, then login into a Jenkins worker node and try exactly the same command line there. Currently, a super long classpath (it fills two and a half screens of mine...) appear on Jenkins for this test suite. This might be the reason why this error only appear on Jenkins, since classpath can be environment sensitive.

**Pull request closed without merging**